### PR TITLE
MVP

### PR DESCRIPTION
**Pull request by oblekta**

### CODE REVIEW

File: game/systems/MVPSystem.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#include <game/components/PositionComponent.h>
+#include <game/states/GameState.h>
+
+namespace game {
+namespace systems {
+    void MVPSystem::update(std::chrono::nanoseconds /*timeStep*/)
+    {
+        auto& crownEntity = m_world->getEntity(m_entityId);
+        auto& crownPosition = m_world->getComponent<components::PositionComponent>(crownEntity).position;
+        engine::ecs::Entity* winner;
```
Comment (notzain) [inline game/systems/MVPSystem.cpp]:
```suggestion
        engine::ecs::Entity* winner = nullptr;
```

File: game/builders/CharacterBuilder.cpp [hunk 2]
```diff
@@ -145,6 +149,19 @@ namespace builders {
 
             m_ecsWorld.addComponent<components::InventoryComponent>(players[i], inventoryComponent);
 
+            if (GameState::hasMVP && currentId == GameState::MVP) {
```
Comment (notzain) [inline game/builders/CharacterBuilder.cpp]: move this to GameState init function?

hasMVP and MVP would not have to be static in that case

**Comment by FabioGNR** [comment]
Crown is cool! Good job.

For the score system we discussed the end screen showing PLAYER -1.
Can you align this with the P1-P4 displayed in the HUD?

File: game/systems/MVPSystem.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        engine::ecs::Entity* winner = nullptr;
+
+        m_world->forEachEntityWith<components::PlayerInputComponent>([&](auto& entity) {
+            auto playerInputComponent = m_world->getComponent<components::PlayerInputComponent>(entity);
+            if (playerInputComponent.controllerId == GameState::MVP) {
+                winner = &entity;
+            }
+        });
+
+        if (winner != nullptr) {
```
Comment (notzain) [inline game/systems/MVPSystem.cpp]: move body to  
```cpp
if (playerInputComponent.controllerId == GameState::MVP) {}
```

then you don't need to keep pointer to the winner

**Comment by oblekta** [comment]
Thanks for the review! I fixed the last things.

**Comment by FabioGNR** [comment]
I don't know how that  clang-tidy error slipped past the punching branch :/

**Pull request merged**